internal/scheduler/legacyscheduler.go

```diff
@@ -0,0 +1,619 @@
+		return jobSchedulingReport, nil
+	}
+
+	// If a NodeDb is provided, try to find a node on which this job can be scheduled.
+	if it.NodeDb != nil {
+		podReport, err := it.selectNodeForPod(ctx, jobId, job)
+		if err != nil {
+			jobSchedulingReport.UnschedulableReason = err.Error()
+			return nil, err
+		}
+		jobSchedulingReport.PodSchedulingReports = append(jobSchedulingReport.PodSchedulingReports, podReport)
+		if podReport.Node == nil {
+			jobSchedulingReport.UnschedulableReason = "pod does not fit on any node"
+			return jobSchedulingReport, nil
+		}
+	}
+
+	return jobSchedulingReport, nil
+}
+
+func uuidFromUlidString(ulid string) (uuid.UUID, error) {
+	protoUuid, err := armadaevents.ProtoUuidFromUlidString(ulid)
+	if err != nil {
+		return uuid.UUID{}, err
+	}
+	return armadaevents.UuidFromProtoUuid(protoUuid), nil
+}
+
+// Check if scheduling this job would exceed per-queue resource limits.
+func (scheduler *LegacyScheduler) exceedsResourceLimits(ctx context.Context, rl schedulerobjects.ResourceList, limits map[string]float64) (bool, string) {
+	for resourceType, limit := range limits {
+		// TODO: Should be computed globally.
+		totalAmount := scheduler.NodeDb.totalResources[resourceType] // TODO: Could be a float computed at init.
+		amountUsedByQueue := rl.Resources[resourceType]
+		if amountUsedByQueue.AsApproximateFloat64()/totalAmount.AsApproximateFloat64() > limit {
+			return true, fmt.Sprintf("scheduling would exceed %s quota", resourceType)
+		}
+	}
+	return false, ""
+}
+
+// Check that this job is at least equal to the minimum job size.
+func (scheduler *LegacyScheduler) jobIsLargeEnough(jobTotalResourceRequests common.ComputeResources) (bool, string) {
+	// TODO: These per-job checks could be expressed as filter functions, e.g., of type
+	// jobsFilterFunc func(*api.Job) bool
+	if len(scheduler.MinimumJobSize) == 0 {
+		return true, ""
+	}
+	for resourceType, quantity := range jobTotalResourceRequests {
+		if limit, ok := scheduler.MinimumJobSize[resourceType]; ok {
+			if limit.Cmp(quantity) == 1 {
+				return false, fmt.Sprintf(
+					"job requests %s %s, but the minimum is %s",
+					quantity.String(), resourceType, limit.String(),
+				)
+			}
+		}
+	}
+	return true, ""
+}
+
+func (scheduler *LegacyScheduler) selectNodeForPod(ctx context.Context, jobId uuid.UUID, job *api.Job) (*PodSchedulingReport, error) {
+	podSpec := podSpecFromJob(job)
+	if podSpec == nil {
+		return nil, errors.New("failed to get pod spec")
+	}
+
+	// Try to find a node for this pod.
+	// Store the report returned by the NodeDb.
+	req := schedulerobjects.PodRequirementsFromPodSpec(podSpec)
+	report, err := scheduler.NodeDb.SelectNodeForPod(jobId, req)
+	if err != nil {
+		return nil, err
+	}
+	return report, nil
+}
+
+type LegacyScheduler struct {
+	SchedulingConfig configuration.SchedulingConfig
+	// Executor for which we're currently scheduling jobs.
+	ExecutorId string
+	// Used for matching pods with nodes.
+	NodeDb *NodeDb
```

Comment:
do we need to export all this outside the module?

Reply:
Let's review it